### PR TITLE
Upgrade macOS image for tests compatibility with numpy v2

### PR DESCRIPTION
**Pull request by echarlaix**
Fix :
https://github.com/huggingface/optimum/actions/runs/11290608247/job/31402830526
https://github.com/huggingface/optimum/actions/runs/11290608260/job/31402830356

### CODE REVIEW

**Comment by IlyasMoutawwakil** [comment]
thanks ! yes many errors with old macos, we might also wanna drop python 3.8 as it's already EOL

**Review by IlyasMoutawwakil — APPROVED** [review]

**Comment by IlyasMoutawwakil** [comment]
of and no more `cp8` on https://download.pytorch.org/whl/nightly/torchaudio/ 👋

File: .github/workflows/test_bettertransformer.yml [hunk 1]
```diff
@@ -16,8 +16,8 @@ jobs:
       fail-fast: false
       matrix:
         python-version: [3.8, 3.9]
```
Comment (IlyasMoutawwakil) [inline .github/workflows/test_bettertransformer.yml:18]:
```suggestion
        python-version: [3.9]
```

File: .github/workflows/test_bettertransformer.yml [hunk 2]
```diff
@@ -16,8 +16,8 @@ jobs:
...
-        os: [ubuntu-20.04, macos-13]
-        exclude: [{ python-version: 3.8, os: macos-13 }]
+        os: [ubuntu-20.04, macos-14]
+        exclude: [{ python-version: 3.8, os: macos-14 }]
```
Comment (IlyasMoutawwakil) [inline .github/workflows/test_bettertransformer.yml:20]:
```suggestion
```

**Comment by echarlaix** [comment]
> thanks ! yes many errors with old macos, we might also wanna drop python 3.8 as it's already EOL

Yes for python 3.8 I wanted to tackle it for all workflows at once but will include it in this one so we're all good, thanks @IlyasMoutawwakil

**Pull request merged**